### PR TITLE
ReignEngine: Implement skeletal animation

### PR DESCRIPTION
**Pull request by kichikuou**
Also adds some debug infra for draw plugins.

### CODE REVIEW

File: src/3d/reign.c [hunk 1]
```diff
@@ -138,29 +159,126 @@ bool RE_instance_load(struct RE_instance *instance, const char *name)
...
+		instance->bone_transforms = xcalloc(instance->model->nr_bones, sizeof(mat4));
 	return !!instance->model;
 }
 
-void RE_debug_print(struct RE_plugin *p)
+bool RE_instance_load_motion(struct RE_instance *instance, const char *name)
+{
+	if (!instance || !instance->model)
+		return false;
+	instance->motion = motion_load(name, instance->model, instance->plugin->aar);
```
Comment (nunuhara) [inline src/3d/reign.c:173]: There is a memory leak here. If instance->motion isn't NULL, it gets overwritten. The leak happens in battles.

File: src/3d/reign.c [hunk 2]
```diff
@@ -138,29 +159,126 @@ bool RE_instance_load(struct RE_instance *instance, const char *name)
...
+	return !!instance->motion;
+}
+
+bool RE_instance_load_next_motion(struct RE_instance *instance, const char *name)
+{
+	if (!instance || !instance->model)
+		return false;
+	instance->next_motion = motion_load(name, instance->model, instance->plugin->aar);
```
Comment (nunuhara) [inline src/3d/reign.c:181]: Same issue as RE_instance_load_motion

File: src/3d/reign.c (hunk 1 above)
Comment (kichikuou) [reply src/3d/reign.c:175]: Good catch, fixed.

File: src/3d/reign.c (hunk 2 above)
Comment (kichikuou) [reply src/3d/reign.c:185]: Fixed.

**Comment by nunuhara** [comment]
Thanks!

**Pull request merged**